index.bs

```diff
@@ -1726,6 +1829,7 @@ To <dfn>attempt to deliver a report</dfn> given an [=attribution report=] |repor
 
 1. Let |url| be the result of executing [=generate an attribution report URL=] on |report|.
 1. Let |data| be the result of executing [=serialize an attribution report=] on |report|.
+1. If |data| is null, return.
```

Comment:
I'm wondering if the later sentence "A user agent may retry this algorithm in the event that there was an error" might need a slight change or to be copied into some of the other algorithms?

Reply:
I think this may be fine as the error has already been handled and data being null is not an error?

Reply:
Ah I think you're right. This algorithm is the main one that calls all the others. I do wonder if being slightly more explicit about what this looks like could be useful, but not super important -- e.g. "If data is null, wait an implementation-defined amount of time and rerun this algorithm." + something similar in processReponse. (Up to you)

One last thing I spotted -- should we be resetting report's delievered bit on startup?

Reply:
Actually updated to return the error to the main algorithm if public key fetching or serialization failed. In this case, the retry statement below that should apply here as well.

Currently the retry logic is not fully spec'ed, so I think it's fine not resetting report's `delivered` bit as we don't want to retry repeatedly.

Reply:
Yeah, I think it's ok not to spec the retry behavior. However, it's a little odd to me that the return path neither changes the delivered bit nor "Queues a task to remove report from the aggregatable report cache." Not sure what the best resolution is, though.

Reply:
yeah, that makes sense. Updated to remove the report from the cache before return.